tests/CoreCLR/build-and-run-test.cmd

```diff
@@ -53,7 +62,14 @@ shift
 goto :GetNextParameter
 
 :RunTest
-%TestFolder%\native\%TestExecutable% %TestParameters%
+
+set CoreRunCommandLine=%CoreRT_CoreCLRRuntimeDir%\CoreRun.exe %TestFolder%\native\%TestFileName%.ni.exe %TestParameters%
```

Comment:
Just in case you're wondering why it didn't work with the set command within the conditional statement, you'd need to use the "setlocal EnableDelayedExpansion" command at the top of the script and replace all percent signs inside the conditional statement with exclamation marks. But I'm completely fine with the way you've done it.